java/code/src/com/redhat/rhn/domain/contentmgmt/PackageFilter.java

```diff
@@ -79,6 +83,18 @@ public boolean test(Package pack) {
         }
     }
 
+    private static String getEvr(String field, String value) {
+        if (field.equals("nevr")) {
+            return value.replaceAll("(.*)-(.*:)?(.*)-(.*)", "$2$3-$4");
+        }
+        else if (field.equals("nevra")) {
+            return value.replaceAll("(.*)-(.*:)?(.*)-(.*)\\.(.*)", "$2$3-$4");
+        }
```

Comment:
One last-minute thought I got: does this play well with the Debian versions? I think it's worth checking it and adding it to the unit tests.

Reply:
Added some test for debian versioning it seems to work fine.